core/command.go

```diff
@@ -94,3 +96,9 @@ func (GoBack) Execute(oldState State) (State, error) {
 		history:   newHistory,
 	}, nil
 }
+
+func (m Mark) Execute(oldState State) (State, error) {
+	newState := copyState(oldState)
+	newState.Folder.Files[newState.Selected].ToggleIsMarked()
```

Comment:
The idea here was (And I'm to blame for not stating it explicitly) That everything related to user interaction would be stored in state. I'd prefer to put a `marked map[*File]string` into the state and every Execution of `Mark` command would add (or remove) the File from the map.

It would be much easier to test. And you would have result ready straight as you finish the interactive mode. What do you think @leonklingele?